server/src/main/java/org/opensearch/index/engine/InternalEngine.java

```diff
@@ -1412,7 +1412,7 @@ private Exception tryAcquireInFlightDocs(Operation operation, int addingDocs) {
         final long totalDocs = indexWriter.getPendingNumDocs() + inFlightDocCount.addAndGet(addingDocs);
         if (totalDocs > maxDocs) {
             releaseInFlightDocs(addingDocs);
-            return new IllegalArgumentException("Number of documents in the index can't exceed [" + maxDocs + "]");
+            return new IllegalArgumentException("Number of documents in shard " + shardId + " can't exceed [" + maxDocs + "]");
```

Comment:
I think we are not delivering the message here we actually, it gives impression that there is a problem with one specific shard but the limit applies to all shards of the index. I would frame it like that:
```
return new IllegalArgumentException("Number of the documents in shard " + shardId + " exceeds the limit of [" + maxDocs + "] per index shard");
```

Reply:
agree @reta 
Update the error message to reflect the same 
```
"Number of documents in shard " + shardId + " exceeds the limit of [" + maxDocs + "] documents per shard"
```